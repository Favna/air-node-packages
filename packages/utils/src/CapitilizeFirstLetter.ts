/**
 * Transforms the first letter to a capital then adds all the rest after it
 * Differs from {@link sentenceCase} in that the casing of the text after the first letter is retained
 * @param str Text to transform
 */
export const capitilizeFirstLetter = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);

export default capitilizeFirstLetter;