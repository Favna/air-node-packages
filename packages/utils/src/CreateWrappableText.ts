/**
 * Inserts a space in any text after every maxLength amount of characters, allowing Material-UI to gracefully wrap it
 * @param text text to split
 * @param maxLength The maximum length of a text section @default 75
 */
export function createWrappableText(text: string, maxLength = 75) {
  const splitRegex = new RegExp(`.{1,${maxLength}}`, 'g');
  const splitContent = text.match(splitRegex);

  if (splitContent) return splitContent.join(' ');

  return text;
}

export default createWrappableText;
