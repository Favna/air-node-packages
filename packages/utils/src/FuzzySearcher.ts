import Fuse from 'fuse.js';

export class FuzzySearcher<O> {
  private readonly object: O[];
  private readonly fuseOptions?: Fuse.IFuseOptions<unknown>;

  constructor(obj: O[], keys: string[], options?: Fuse.IFuseOptions<unknown>) {
    this.object = obj;
    this.fuseOptions = {
      threshold: 0.2,
      ...options,
      keys
    };
  }

  public runFuzzy(query: string) {
    const locquery = query.toLowerCase();

    const fuzzyFuze = new Fuse(this.object, this.fuseOptions);

    return fuzzyFuze.search<O>(locquery);
  }
}
