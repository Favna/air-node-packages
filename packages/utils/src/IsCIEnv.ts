/** Validates whether the current environment is a CI environment */
export const isCIEnv = Boolean(process.env.CI);

export default isCIEnv;