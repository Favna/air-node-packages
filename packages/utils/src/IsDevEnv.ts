/** Validates whether the current environment is either development or test */
export const isDevEnv = process.env.NODE_ENV === 'development' ? true : process.env.NODE_ENV === 'test';

export default isDevEnv;
