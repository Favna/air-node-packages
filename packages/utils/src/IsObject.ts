/**
 * Verify if the input is an object literal (or class).
 * @param input The object to verify
 */
export const isObject = (input: unknown): boolean => {
  return input ? (input as object).constructor === Object : false;
};

export default isObject;