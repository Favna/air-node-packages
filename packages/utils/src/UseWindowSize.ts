import { useState, useEffect } from 'react';

/**
 * React hook to get the current window size
 * Triggers on window resize.
 *
 * Use with {@link https://github.com/xnimorz/use-debounce | use-debounce} to ensure it is not triggered instantly!!
 *
 * @example
 * ```typescript
 * import { useDebounce } from 'use-debounce';
 *
 * const [ width, height ] = useDebounce(useWindowSize(), 1000);
 *
 * useEffect(() => {
 *   // Do something here
 * }, [ width, height ]);
 * ```
 */
export function useWindowSize() {
  const getSize = (): [number, number] => {
    return [window.innerWidth, window.innerHeight];
  };

  const [windowSize, setWindowSize] = useState(getSize);

  useEffect(() => {
    function handleResize() {
      setWindowSize(getSize());
    }

    window.addEventListener('resize', handleResize);

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowSize;
}
