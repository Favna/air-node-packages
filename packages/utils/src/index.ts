export * from './CapitilizeFirstLetter';
export * from './CreateWrappableText';
export * from './CutText';
export * from './IsDevEnv';
export * from './IsTestEnv';
export * from './ObjectHasProperty';
export * from './SentenceCase';
export * from './SnakeToCamelCase';
export * from './SplitText';
export * from './StableTableSort';
export * from './TimeUtils';
export * from './UseWindowSize';