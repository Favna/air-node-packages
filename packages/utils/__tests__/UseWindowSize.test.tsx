import React, { FC } from 'react'; // eslint-disable-line @typescript-eslint/no-unused-vars
import { useWindowSize } from '../src/UseWindowSize';
import { shallow } from 'enzyme';

interface ComponentProps {
  name: string;
}

export const Component: FC<ComponentProps> = props => {
  const [width, height] = useWindowSize();

  return (
    <div>
      {width} {height} {props.name}
    </div>
  );
};

test('should have width and height', () => {
  const component = shallow(<Component name='Robin Hood' />);

  expect(component.text()).toStrictEqual('1024 768 Robin Hood');

  expect(component).toMatchSnapshot();
});
