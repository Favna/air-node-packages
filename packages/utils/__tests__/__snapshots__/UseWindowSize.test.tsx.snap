// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have width and height 1`] = `
<div>
  1024
   
  768
   
  Robin Hood
</div>
`;
